### DISCUSSION TITLE
Visibility of the project and its status

### DISCUSSION THREAD

**Discussion by buhtz** [WeblateOrg/weblate#12205] [General] [answered]
Hello Michal,

follow up to [your comment](https://github.com/WeblateOrg/weblate/pull/12199#issuecomment-2270465447).

> The bot comments on missing code coverage ... If you don't see a value in that

I was not clear about what I meant. Code coverage is important but not to the public. The bot is not only useless but annoying. Code coverage is of course not. The primary purpose of the "Files changed" view in a PR is to do reviewing the code. The bot messages to interrupt that because the code is harder to read. I am just concerned about the review process.

Your project is also a good candidate to attract beginner contributors. But you need better documentation and guidance for it. Yes, this cots more time than it saves in the first place. But on a long run some of the contributors will stick to the project and also other contributors will see how gentle you guide beginners and being also attracted to the project.

> We don't need guidance on where more work can be done

You get me wrong. Your todo list was not the topic.

> Weblate is now healthier than ever

I believe you. But from an outside view it is not obvious. You should be more transparent about it.
An outside see only one account committing and PRs not reviewed. That is not considered as healthy.
There might be good reasons. But be transparent about them. Write about it. Describe your team and your workflow. Your motivation. This will attract more contributors.
One real "problem" is the company behind. Some potential contributors might think that they won't contribute to a company. They have money so they should pay developers and contributors. It is not my opinion but a way some might think.
It is also not clear what happens to your donations? How many? From who? Any other companies doing donations? What are you doing with it?

Transparency grows the community and attract contributors.

This is all about marketing. Your website looks fancy. But there is only one person doing commits and merging unreviewed PRs. That does not fit together and do not grow trust into a project.

What is about Codeberg.org? You never mention that on your website. Having them in the back is really a value from a marketing perspective because these folks have a very good reputation.

**Comment by nijel** [comment]
I'll split the response into two threads, let's start with the code coverage:

> I was not clear about what I meant. Code coverage is important but not to the public. The bot is not only useless but annoying. Code coverage is of course not. The primary purpose of the "Files changed" view in a PR is to do reviewing the code. The bot messages to interrupt that because the code is harder to read. I am just concerned about the review process.

Lack of code coverage is part of the code review. The linting messages will appear as well. Hiding all the annotations is one key press or two clicks away if you would rather not see them (see https://docs.codecov.com/docs/github-checks#hiding-annotations-in-the-files-view).

**Comment by nijel** [comment, marked as answer]
> Your project is also a good candidate to attract beginner contributors. But you need better documentation and guidance for it. Yes, **this cots more time** than it saves in the first place. But on a long run some of the contributors will stick to the project and also other contributors will see how gentle you guide beginners and being also attracted to the project.

Emphasis mine...
 
> > We don't need guidance on where more work can be done
> 
> You get me wrong. Your todo list was not the topic.

...so it's not about a todo list, but it's about spending time on things you see important. Honestly, I don't see a difference between these two. It's definitely a thing that should be done, but there are many other things like that I see more important.

> 
> > Weblate is now healthier than ever
> 
> I believe you. But from an outside view it is not obvious. You should be more transparent about it. 

> An outside see only one account committing and PRs not reviewed. That is not considered as healthy. There might be good reasons. But be transparent about them. Write about it. Describe your team and your workflow. Your motivation. This will attract more contributors.

That seems like a quite limited outside view. Let's look at yesterday's commits, there are https://github.com/WeblateOrg/weblate/commit/7722ae6a3bbfd41b821f7f03baaab40b17fd1042 and https://github.com/WeblateOrg/weblate/commit/579e27c622fef83bac3b5c5dbf23e9021be04faf. Overall, 13 persons (besides me) contributed code for the upcoming release.

> One real "problem" is the company behind. Some potential contributors might think that they won't contribute to a company. They have money so they should pay developers and contributors. It is not my opinion but a way some might think.

I work full-time on Weblate and I need money to live. That's why there is a company to fund Weblate which offers commercial services. This is actually what allowed me to work full-time on Weblate years ago, and not the donations.

> It is also not clear what happens to your donations? How many? From who? Any other companies doing donations? What are you doing with it?

Some time ago, I decided to use donations to fund libraries and tools we build on. These are usually in a much more money constrained situation than we are. But honestly, there are not that much of donations. Companies typically purchase services and not give donations.

> Transparency grows the community and attract contributors.
> 
> This is all about marketing. Your website looks fancy. But there is only one person doing commits and merging unreviewed PRs. That does not fit together and do not grow trust into a project.

We could definitely do more in marketing, that's why we've got [open positions](https://weblate.org/careers/) so that @orangesunny can spend more time on that part of his job.

> What is about Codeberg.org? You never mention that on your website. Having them in the back is really a value from a marketing perspective because these folks have a very good reputation.

We're not affiliated with them, and I learned about that from reported issues. Maybe they could enable discoverability of their Weblate instance on https://weblate.org/discover/? See https://docs.weblate.org/en/latest/admin/support.html#discover-weblate

**Reply by buhtz** [reply]
> Maybe they could enable discoverability of their Weblate instance on https://weblate.org/discover/? See https://docs.weblate.org/en/latest/admin/support.html#discover-weblate

I contacted the Codeberg folks: https://codeberg.org/Codeberg/Community/issues/1626
